src/transforms/bin.js

```diff
@@ -74,7 +83,7 @@ function binn(
   gx, // optionally group on x (exclusive with bx and gy)
   gy, // optionally group on y (exclusive with by and gx)
   {
-    data: reduceData = reduceIdentity,
+    data: reduceData = reduceIdentity, // TODO avoid materializing when unused?
```

Comment:
Not a single one of the tests seems to be using the return value of reduceIdentity.reduce.